## Tests Concerning the Mean Difference

### Load Data

In [71]:
library(tidyverse)
dat = read.table('data/shhs1.txt', header=T, na.strings=".")

In [72]:
smry=dat %>% group_by(gender) %>% summarise(mean=mean(rdi4p), sd=sd(rdi4p), n=n())
smry

gender mean      sd       n   
1 0       6.154055 10.18041 3039
2 1      11.404863 14.00619 2765

In [31]:
nx = smry$n[1]
ny = smry$n[2]
mux = smry$mean[1]
muy = smry$mean[2]
sdx = smry$sd[1]
sdy = smry$sd[2]
paste(c(nx, ny, mux, muy, sdx, sdy))

[1] "3039"             "2765"             "6.15405531155413" "11.4048633909056"
[5] "10.1804130079805" "14.0061907811066"

### The estimate for mean difference

In [49]:
estimate = abs(( mux - muy ))
estimate

[1] 5.250808

### Standard error

In [33]:
se = sqrt(sdx^2 / nx + sdy^2 / ny)
se

[1] 0.3241178

### Degrees of freedom

In [68]:
# See https://academic.oup.com/beheco/article/17/4/688/215960
u = (sdy/sdx)^2
df = round((1/nx + u/ny)^2/( 1/(nx^2 * (nx - 1)) + u^2/(ny^2 * (ny - 1)) ), 3)
df

[1] 5007.234

### The *t*-test statistic

In [35]:
ts = estimate  / se  # pooled se sqrt(var_1/nx + var_2 / ny)
ts

[1] 16.20031

### Conclusion

Since 16.2 is much higher than $t_{df, 1-\alpha / 2 }$ (for large $df$ this aproximates $Z_{1 - \alpha/2}$  $\approx$ 1.96 )

### Gosset's *t* quantile for 95% ci

In [60]:
# 95 + (100-95)/5 = 97.5
q975 = qt( .975, df )
q975

[1] 1.959967

In [61]:
round( estimate - q975 * sqrt( sdx^2 / nx  + sdy^2 / ny ), 3)

[1] 4.616

In [62]:
round( estimate + q975 * sqrt( sdx^2 / nx  + sdy^2 / ny ), 3)

[1] 5.886

## Welch's Approximation

In [70]:
t.test(dat$rdi4p ~ dat$gender, var.equal=F)


	Welch Two Sample t-test

data:  dat$rdi4p by dat$gender
t = -16.2, df = 5007.2, p-value < 2.2e-16
alternative hypothesis: true difference in means is not equal to 0
95 percent confidence interval:
 -5.886221 -4.615395
sample estimates:
mean in group 0 mean in group 1 
       6.154055       11.404863 


## P-Values
Consider an experiment of 10 coin flips that resulted in 2 heads. So the hypotheses are: $H_{0}$: p = .5 and $H_{a}$: p $\neq$ .5. The following code will calculate p-value from the sample statistic for a binomial distribution with n = 10 and $p_{0}$ = .5.

In [82]:
pbinom(2, 10, .5, lower.tail=T)

[1] 0.0546875

### Rephrase the test
 but now we will perform testing for the result of getting 8 heads or more.

In [83]:
pbinom(7, 10, .5, lower.tail=F)

[1] 0.0546875